src/test/ui/issues/issue-49824.stderr

```diff
@@ -1,11 +1,14 @@
 error: captured variable cannot escape `FnMut` closure body
   --> $DIR/issue-49824.rs:4:9
    |
+LL |       let mut x = 0;
+   |           ----- variable defined here
 LL |       || {
    |        - inferred to be a `FnMut` closure
 LL | /         || {
 LL | |
 LL | |             let _y = &mut x;
+   | |                           - variable captured here
 LL | |         }
    | |_________^ returns a closure that contains a reference to a captured variable, which then escapes the closure body
```

Comment:
is this the "prime example" we are targetting?

Reply:
The original issue was an async-await + closure snippet, which I included in `src/test/ui/async-await/issue-69446-fnmut-capture.rs`.

But yes, I would expect this file to represent the most common case for this PR.